src/WebJobs.Script/Host/ScriptHostManager.cs

```diff
@@ -44,16 +47,18 @@ public class ScriptHostManager : IScriptHostEnvironment, IDisposable
         private ILogger _logger;
 
         private ScriptSettingsManager _settingsManager;
-        private AutoRecoveringFileSystemWatcher _scriptFileWatcher;
         private CancellationTokenSource _restartDelayTokenSource;
 
         public ScriptHostManager(ScriptHostConfiguration config, IScriptEventManager eventManager = null, IScriptHostEnvironment environment = null)
             : this(config, ScriptSettingsManager.Instance, new ScriptHostFactory(), eventManager, environment)
         {
             if (config.FileWatchingEnabled)
             {
-                _scriptFileWatcher = new AutoRecoveringFileSystemWatcher(config.RootScriptPath);
-                _scriptFileWatcher.Changed += OnScriptFileChanged;
+                _fileEventSource = new FileWatcherEventSource(EventManager, "ScriptFiles", config.RootScriptPath);
+
+                EventManager.OfType<FileEvent>()
+                    .Where(f => string.Equals(f.Source, "ScriptFiles", StringComparison.Ordinal))
```

Comment:
We can. The source here is pretty specific to file watchers (it's part of the `FileEvent`), so hopefully, its use will be limited. We could create the extension if we feel it is needed, though.